makemkv/types.py

```diff
@@ -57,18 +57,18 @@ class Title(TypedDict, total=False):
     information: str
     length: str  # hh:mm:ss
     name: str  # eg. "title"
-    metadata_langcode: str  # eg. "eng"
+    metadata_langcode: str  # two-letter ISO 639-1 code if it exists, eg. "en"
     metadata_language: str  # eg. "English"
     segments_count: int  # eg. 5
-    segments_map: str  # eg. "174,175,175,175,448"
+    segments_map: Iterable[int]  # eg. [174,175,175,175,448]
```

Comment:
Since we always return a list, this should be `list[str]` (this also applies to `streams`, `drives` and `titles`).

I noticed that this causes some problems with the validation in `_is_valid_typeddict_item`, so I'll open a PR against your branch to fix this.

Reply:
Of course it should be `list[int]`, not  `list[str]`. See my PR.

Reply:
it only ran with list not list[int], it fails at isinstance

Reply:
your changes resolved this thanks!